### DISCUSSION TITLE
Models hashes changed

### DISCUSSION THREAD

**Discussion by Melyns** [lllyasviel/stable-diffusion-webui-forge#1477] [General]
SDXL & Lora Models suddenly changed hash recently. I noticed that because images i was uploading to civitai were no longer showing resources i used. I tried CRC32 hashes of models i have downloaded from civitai, & the ones at civitai but they aren't the same. Example:

ZavyChromaXL - 71F4AC92 (civitai)
ZavyChromaXL - 92ACF471 (local)
JuggernautXL - AABEAB5C (civitai)
JuggernautXL - 5AD33354 (local)

Models are not corrupted or anything & function as expected. im not saying or blaming forge for this but i don't really know what happened. One thing i remember doing is downloading & using this extension for gnerating preview images & model info cards https://github.com/BlafKing/sd-civitai-browser-plus.git so could it possibly be the reason?

Can i change the hashes manually without having to redownload?

if you have an explanation for how could hashes change without file corruption please i'd like to understand

**Comment by DenOfEquity** [comment]
It's hashing the filename, not the file contents. I think the change was made when model loading behaviour was changed, possibly an abundance of caution<sup>TM</sup>.
The hash used is SHA256.

**Reply by trunksn1** [reply]
Is the hash stored anywhere or is it get generated every time? 

could it be possible to "inject" the correct hash, bypassing the hashing function of forge?

**Comment by mistermcluvin** [comment]
I reported this last week. 
https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1242#discussioncomment-10411649
I think Civitai is looking for the models AutoV2 Hash they have listed.

**Comment by JEAYNE** [comment]
> ZavyChromaXL - 71F4AC92 (civitai)
> ZavyChromaXL - 92ACF471 (local)

I'm facing the same problem. Here an example:
```
chroma_v41LowStepRl.safetensors - B49BFC57 (civitai)
chroma_v41LowStepRl.safetensors - 57fc9bb4 (local)
```

**But there seems to be a certain logic to it**: Reading the CRC32 of Civitail from left to right, byte by byte, is identical to reading the local CRC32 from right to left, byte by byte. (maybe a little/big endian issue on civitai side ?)

Notes:
- I computed the local CRC32 with several tools/libs
- The hashes (CRC32, SHA256...) correspond to the hashes of the file **contents**.
- The SHA256 displayed by civitai are ok.
